packages/amplify-e2e-core/src/utils/sdk-calls.ts

```diff
@@ -401,42 +403,127 @@ export const getSSMParameters = async (region: string, appId: string, envName: s
     .promise();
 };
 
+export const getSSMParametersFunctionPrefix = async (
+  region: string,
+  appId: string,
+  envName: string,
+  funcName: string,
+  parameterNames: string[],
+) => {
+  const ssmClient = new SSM({ region });
+  if (!parameterNames || parameterNames.length === 0) {
+    throw new Error('no parameterNames specified');
+  }
+  return await ssmClient
+    .getParameters({
+      Names: parameterNames.map(name => `/amplify/${appId}/${envName}/AMPLIFY_function_${funcName}_${name}`),
+      WithDecryption: true,
+    })
+    .promise();
+};
+
+export const getAllSSMParamatersFromAppId = async (appId: string, region: string): Promise<Array<string>> => {
+  const ssmClient = new SSM({ region });
+  const retrievedParameters: Array<string> = [];
+  let recievedNextToken = '';
+  do {
+    const ssmArgument = getSsmSdkParametersGetParametersByPath(appId, recievedNextToken);
+    const data = await ssmClient.getParametersByPath(ssmArgument).promise();
+    retrievedParameters.push(...data.Parameters.map(returnedParameter => returnedParameter.Name));
+    recievedNextToken = data.NextToken;
+  } while (recievedNextToken);
+  return retrievedParameters;
+};
+
+export const expectParametersOptionalValue = async (
```

Comment:
Is there a more descriptive name that can be used? I don't understand the optional value part, because all the function parameters are required.

Reply:
The optional value part means that you can pass in a key-value pair to expect, or just a key. So the value is optional. This is for parameters where the value cannot be predicted.